partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnGt.kt

```diff
@@ -4,201 +4,138 @@
-    parameters = arrayOf(
-        Parameter("lhs", PType.string()),
-        Parameter("rhs", PType.string()),
-    ),
-
-) { args ->
-    val lhs = args[0]
-    val rhs = args[1]
-    Datum.bool(lhs.string > rhs.string)
-}
-
-internal val Fn_GT__DATE_DATE__BOOL = Function.static(
-
-    name = "gt",
-    returns = PType.bool(),
-    parameters = arrayOf(
-        Parameter("lhs", PType.date()),
-        Parameter("rhs", PType.date()),
-    ),
-
-) { args ->
-    val lhs = args[0]
-    val rhs = args[1]
-    Datum.bool(lhs.date > rhs.date)
-}
-
-internal val Fn_GT__TIME_TIME__BOOL = Function.static(
-
-    name = "gt",
-    returns = PType.bool(),
-    parameters = arrayOf(
-        Parameter("lhs", PType.time(6)),
-        Parameter("rhs", PType.time(6)),
-    ),
-
-) { args ->
-    val lhs = args[0]
-    val rhs = args[1]
-    Datum.bool(lhs.time > rhs.time)
-}
-
-internal val Fn_GT__TIMESTAMP_TIMESTAMP__BOOL = Function.static(
-
-    name = "gt",
-    returns = PType.bool(),
-    parameters = arrayOf(
-        Parameter("lhs", PType.timestamp(6)),
-        Parameter("rhs", PType.timestamp(6)),
-    ),
-
-) { args ->
-    val lhs = args[0]
-    val rhs = args[1]
-    Datum.bool(lhs.timestamp > rhs.timestamp)
-}
-
-internal val Fn_GT__BOOL_BOOL__BOOL = Function.static(
-
-    name = "gt",
-    returns = PType.bool(),
-    parameters = arrayOf(
-        Parameter("lhs", PType.bool()),
-        Parameter("rhs", PType.bool()),
-    ),
-
-) { args ->
-    val lhs = args[0]
-    val rhs = args[1]
-    Datum.bool(lhs.boolean > rhs.boolean)
+import org.partiql.value.util.compareTo
+
+internal object FnGt : DiadicComparisonOperator("gt") {
+
+    init {
+        fillTable()
+    }
+
+    override fun getComparison(lhs: Number, rhs: Number): Boolean {
+        return lhs > rhs
+    }
+
+    override fun getTinyIntInstance(tinyIntLhs: PType, tinyIntRhs: PType): Function.Instance {
+        return basic(PType.bool(), PType.tinyint()) { args ->
+            val lhs = args[0].byte
+            val rhs = args[1].byte
+            Datum.bool(lhs > rhs)
+        }
+    }
+
+    override fun getSmallIntInstance(smallIntLhs: PType, smallIntRhs: PType): Function.Instance {
+        return basic(PType.bool(), PType.smallint()) { args ->
+            val lhs = args[0].short
+            val rhs = args[1].short
+            Datum.bool(lhs > rhs)
+        }
+    }
+
+    override fun getIntegerInstance(integerLhs: PType, integerRhs: PType): Function.Instance {
+        return basic(PType.bool(), PType.integer()) { args ->
+            val lhs = args[0].int
+            val rhs = args[1].int
+            Datum.bool(lhs > rhs)
+        }
+    }
+
+    override fun getBigIntInstance(bigIntLhs: PType, bigIntRhs: PType): Function.Instance {
+        return basic(PType.bool(), PType.bigint()) { args ->
+            val lhs = args[0].long
+            val rhs = args[1].long
+            Datum.bool(lhs > rhs)
+        }
+    }
+
+    // TODO: Update
```

Comment:
Guessing this TODO is related to the upcoming change for updating `numeric` representation? If so, could add a bit more context in the TODO comments in this and the other comparison functions.